File: bench/micro/form_vec_get.go
// Go equivalent of form_vec_get.ap.
// Populate (outside timing) then time bounds-checked indexed
// reads. Go's `v[i]` is bounds-checked by default — closest
// analog to Aperio's `get(i) or raise`.
package main

import (
	"fmt"
	"time"
)

func main() {
	iters := 200000
	v := make([]int, 0, iters)
	for i := 0; i < iters; i++ {
		v = append(v, i)
	}

	t0 := time.Now()
	acc := 0
	for j := 0; j < iters; j++ {
		acc = v[j]
	}
	elapsed := time.Since(t0).Nanoseconds()

	fmt.Printf("iters=%d\n", iters)
	fmt.Printf("acc=%d\n", acc)
	fmt.Printf("elapsed_ns=%d\n", elapsed)
}

File: bench/micro/form_vec_get.py
"""Python equivalent of form_vec_get.ap.
Populate (outside timing) then time indexed reads.
"""

import time

iters = 200_000
v = [i for i in range(iters)]

t0 = time.monotonic_ns()
acc = 0
for j in range(iters):
    acc = v[j]
elapsed = time.monotonic_ns() - t0
print(f"iters={iters}")
print(f"acc={acc}")
print(f"elapsed_ns={elapsed}")
